File: typescript/base-hmi-types/src/screens/widgets/HmiScaleWidgetBase.ts
import { HmiColor, HmiFont, HmiProperty } from "../base.js";
import { HmiWidgetBase } from "./HmiWidgetBase.js";
import { HmiThreshold, HmiThresholdValueMode } from "./HmiThreshold.js";
import { HmiTickDirection } from "./HmiTickDirection.js";

export abstract class HmiScaleWidgetBase extends HmiWidgetBase {
  configuredThresholdCount?: HmiProperty<number>;
  thresholdValueMode?: HmiProperty<HmiThresholdValueMode>;
  readonly thresholds: HmiThreshold[] = [];
  value?: HmiProperty<number>;
  fillLevel?: HmiProperty<number>;
  showFillLevel?: HmiProperty<boolean>;
  showScale?: HmiProperty<boolean>;
  drawInsideFrame?: HmiProperty<boolean>;
  showTickLabels?: HmiProperty<boolean>;
  tickLabelFieldLength?: HmiProperty<number>;
  tickLabelDecimalPlaces?: HmiProperty<number>;
  useAutoScaling?: HmiProperty<boolean>;
  showLimitRanges?: HmiProperty<boolean>;
  beginValue?: HmiProperty<number>;
  endValue?: HmiProperty<number>;
  originValue?: HmiProperty<number>;
  divisionCount?: HmiProperty<number>;
  subDivisionCount?: HmiProperty<number>;
  tickDirection?: HmiProperty<HmiTickDirection>;
  barMode?: HmiProperty<number>;
  scaleMode?: HmiProperty<number>;
  orientation?: HmiProperty<number>;
  showValue?: HmiProperty<boolean>;
  valuePosition?: HmiProperty<number>;
  labelColor?: HmiProperty<HmiColor>;
  scaleBackgroundColor?: HmiProperty<HmiColor>;
  scaleForegroundColor?: HmiProperty<HmiColor>;
  tickColor?: HmiProperty<HmiColor>;
  labelFont?: HmiFont;
}

File: src/BaseHmiTypes/Screens/Widgets/HmiScaleWidgetBase.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Widgets;

public abstract class HmiScaleWidgetBase : HmiWidgetBase
{
    public HmiProperty<int>? ConfiguredThresholdCount { get; set; }

    public HmiProperty<HmiThresholdValueMode>? ThresholdValueMode { get; set; }

    public IList<HmiThreshold> Thresholds { get; } = new List<HmiThreshold>();

    public HmiProperty<double>? Value { get; set; }

    public HmiProperty<double>? FillLevel { get; set; }

    public HmiProperty<bool>? ShowFillLevel { get; set; }

    public HmiProperty<bool>? ShowScale { get; set; }

    public HmiProperty<bool>? DrawInsideFrame { get; set; }

    public HmiProperty<bool>? ShowTickLabels { get; set; }

    public HmiProperty<bool>? UseAutoScaling { get; set; }

    public HmiProperty<bool>? ShowLimitRanges { get; set; }

    public HmiProperty<double>? BeginValue { get; set; }

    public HmiProperty<double>? EndValue { get; set; }

    public HmiProperty<double>? OriginValue { get; set; }

    public HmiProperty<int>? DivisionCount { get; set; }

    public HmiProperty<int>? SubDivisionCount { get; set; }

    public HmiProperty<HmiTickDirection>? TickDirection { get; set; }

    public HmiProperty<int>? BarMode { get; set; }

    public HmiProperty<int>? ScaleMode { get; set; }

    public HmiProperty<int>? Orientation { get; set; }

    public HmiProperty<bool>? ShowValue { get; set; }

    public HmiProperty<int>? ValuePosition { get; set; }

    public HmiProperty<HmiColor>? LabelColor { get; set; }

    public HmiProperty<HmiColor>? ScaleBackgroundColor { get; set; }

    public HmiProperty<HmiColor>? ScaleForegroundColor { get; set; }

    public HmiProperty<HmiColor>? TickColor { get; set; }

    public HmiFont? LabelFont { get; set; }
}
